app/models/category_channel.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CategoryChannel < ChatChannel
+  alias_attribute :category, :chatable
+
+  delegate :read_restricted?, to: :category
+
+  def category_channel?
+    true
+  end
+  alias public_channel? category_channel?
```

Comment:
should we mark some of these calls as deprecated?

Reply:
Not sure as those methods are used at different places in different contexts. Probably at some point we could (should?) simplify the code which uses them.